moveit_cartesian_plan_plugin/src/generate_cartesian_path.cpp

```diff
@@ -40,13 +41,68 @@ void GenerateCartesianPath::init()
         - Joint Model group which are necessary for checking if Way-Point is outside the IK Solution
         .
   */
-  moveit_group_ = MoveGroupPtr(new move_group_interface::MoveGroup("manipulator"));
-  kinematic_state = moveit::core::RobotStatePtr(moveit_group_->getCurrentState());
-  kinematic_state->setToDefaultValues(); 
+  selected_plan_group = 0;
+  robot_model_loader = RobotModelLoaderPtr(new robot_model_loader::RobotModelLoader("robot_description"));
+  kmodel = robot_model_loader->getModel();
+  //kmodel = robot_model::RobotModelConstPtr(new moveit::core::RobotModel(robot_model_loader->getModel()));
+
+  //group_names = kmodel->getJointModelGroupNames();
+
+  end_eff_joint_groups = kmodel->getEndEffectors();
   
 
-  const robot_model::RobotModelConstPtr &kmodel = kinematic_state->getRobotModel();
-  joint_model_group = kmodel->getJointModelGroup("manipulator");
+  ROS_INFO_STREAM("size of the end effectors is: "<<end_eff_joint_groups.size());
+
+  if (end_eff_joint_groups.empty())
```

Comment:
Out of curiosity, why is there different methods for handling end-effectors vs. standard kinematic chains?  We don't use end-effectors in ROS-I, mostly out of ignorance about how they are useful.